frontend/src/utils/fetch-with-timeout.ts

```diff
@@ -29,10 +29,16 @@ export const fetchWithTimeout = async (
       signal: controller.signal,
     });
     if (!res.ok) {
-      throw new HTTPError(
-        fetchErrorMessage ?? `Something went wrong requesting at ${resource}`,
-        res.status,
-      );
+      await res.json().then(r => {
+        // eslint-disable-next-line no-console
+        console.log(r);
+        throw new HTTPError(
+          r.detail ??
+            fetchErrorMessage ??
+            `Something went wrong requesting at ${resource}`,
+          res.status,
+        );
+      });
```

Comment:
Nice, I like that we're now logging the response on errors. I think this response would be useful to include in the HTTPError class. If we add a errorBody property to HTTPError and pass in r as the third parameter when initializing HTTPError here, our error log in the the catch statement below will show the response body log in the same message as the other details of the error. I think that'll be easier to parse when debugging in the future